core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java

```diff
@@ -3167,6 +3172,412 @@ public void testRegisterExistingTable() {
+
+    table
+        .manageSnapshots()
+        .createBranch(branchName, firstSnapshot.snapshotId())
+        .setMinSnapshotsToKeep(branchName, minSnapshotsToKeep)
+        .setMaxSnapshotAgeMs(branchName, maxSnapshotAgeMs)
+        .commit();
+
+    table.newFastAppend().toBranch(branchName).appendFile(FILE_B).commit();
+    table.newFastAppend().toBranch(branchName).appendFile(FILE_C).commit();
+
+    // Wait for snapshots to age
+    Uninterruptibles.sleepUninterruptibly(1100, TimeUnit.MILLISECONDS);
+
+    // Expire snapshots
+    table.expireSnapshots().cleanExpiredMetadata(true).commit();
+
+    // Verify retention policy was enforced
+    Map<String, SnapshotRef> refs = table.refs();
+    SnapshotRef branchRef = refs.get(branchName);
+    assertThat(branchRef).isNotNull();
+
+    // The branch should still exist and point to a recent snapshot
+    Snapshot currentBranchSnapshot = table.snapshot(branchRef.snapshotId());
+    assertThat(currentBranchSnapshot).isNotNull();
+
+    String tagName = "short-lived-tag";
+    long maxRefAgeMs = 1000L; // 1 second
+
+    table
+        .manageSnapshots()
+        .createTag(tagName, firstSnapshot.snapshotId())
+        .setMaxRefAgeMs(tagName, maxRefAgeMs)
+        .commit();
+
+    // Verify tag exists
+    refs = table.refs();
+    assertThat(refs).containsKey(tagName);
+
+    // Wait for tag to age
+    Uninterruptibles.sleepUninterruptibly(1100, TimeUnit.MILLISECONDS);
+
+    // Expire snapshots (this should also expire the tag)
+    table.expireSnapshots().cleanExpiredMetadata(true).commit();
+
+    // Verify tag was expired
+    refs = table.refs();
+    assertThat(refs).doesNotContainKey(tagName);
+
+    String nonExistentBranch = "non-existent-branch";
+    assertThatThrownBy(
+            () -> table.manageSnapshots().setMinSnapshotsToKeep(nonExistentBranch, 5).commit())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining("Branch does not exist: %s", nonExistentBranch);
+
+    String tagName2 = "test-tag";
+    table.manageSnapshots().createTag(tagName2, firstSnapshot.snapshotId()).commit();
+
+    // Attempt to set branch-specific retention policy on tag should fail
+    assertThatThrownBy(() -> table.manageSnapshots().setMinSnapshotsToKeep(tagName2, 5).commit())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining("Tags do not support setting minSnapshotsToKeep");
+  }
+
+  @Test
+  public void testMultiReferenceManagement() {
+    C catalog = catalog();
+
+    if (requiresNamespaceCreate()) {
+      catalog.createNamespace(NS);
+    }
+
+    Table table = catalog.buildTable(TABLE, SCHEMA).create();
+    table.newFastAppend().appendFile(FILE_A).commit();
+    Snapshot baseSnapshot = table.currentSnapshot();
+
+    String branch1 = "feature-1";
+    String branch2 = "feature-2";
+    String branch3 = "hotfix";
+    String tag1 = "v1.0";
+    String tag2 = "v1.1";
+    String tag3 = "release-candidate";
```

Comment:
Very minor point: These names are really like git like release tags/branches, it'd be nice to use more realistic names for data oriented branches and tags (i.e. end of month tags , audit branches etc). No big deal though just something that caught my eye :)

Reply:
Done. :) Check names now